src/gradient.py

```diff
@@ -131,45 +130,53 @@ def fool_acc(self, max_iter=None, verbose=True, grid=None,random_state=None, met
             random_state=random_state,
             method="ale++",
         )
+        self._X_changed = X_changed
 
         target = tf.convert_to_tensor(self.result_explanation['original'])
         y = tf.convert_to_tensor(self._y, dtype=tf.float32)
-        loss = self.calculate_alepp_loss(self._X, y, target, reg_factor)
+        raw_loss, loss = self.calculate_alepp_loss(self._X, self._X_changed, y, reg_factor)
         self.iter_losses['iter'].append(0)
+        self.iter_losses['raw_loss'] = [raw_loss]
         self.iter_losses['loss'].append(loss)
 
         self.explainer.model.evaluate(self._X, self._y)
-        optimizer = tf.keras.optimizers.Adam(learning_rate=self.params['learning_rate'])
+        optimizer = tf.keras.optimizers.Adam()
```

Comment:
Czemu wywalony learning rate?

Reply:
Z tego co pamiętam to jest learning rate do innego procesu uczenia; możliwe że to do trenowania ataku, podczas gdy tutaj trenujemy sam model.